projects/control-service/CONTRIBUTING.md

```diff
@@ -1,55 +1,131 @@
 
-## Build and run the Control Plane service
-run `./gradlew tasks` to see all possible tasks and descriptions
+```properties
+datajobs.docker.registryUsername=<your-github-username>
+datajobs.docker.registryPassword=<your-personal-access-token>
+​
+datajobs.deployment.k8s.kubeconfig=<path-to-kubeconfig>
 
-### Build
+datajobs.git.url=github.com/<your-github-username/<your-github-repo>.git
+datajobs.git.username=<your-github-username>
+datajobs.git.password=<your-personal-access-token>
 
-This will build project and create docker image
-```bash
-  ./cicd/build.sh
+datajobs.proxy.repositoryUrl=ghcr.io/<your-github-username/your-github-repo>
+datajobs.git.read.write.username=<your-github-username>
+datajobs.git.read.write.password=<your-personal-access-token>
 ```
 
-Or (without creating docker image)
+**Build the project**
 
 ```bash
 ./gradlew -p ./model build publishToMavenLocal --info --stacktrace
 ./gradlew build jacocoTestReport -x integrationTest --info --stacktrace
 ```
 
+**Run the project**
+
+```bash
+./gradlew :pipelines_control_service:docker  --info --stacktrace -Pversion=dev
+./gradlew :pipelines_control_service:bootRun
+```
+
+Note: You can also build and run using IntelliJ by running the
+`com/vmware/taurus/ServiceApp.java` file
+
 ### Setup IntelliJ:
-* Install Lombok Plugin
-* Import control-service project from existing sources in IntelliJ
-* Link Gradle Project to control-service/projects/settings.gradle
+
+* Import `control-service/projects` in IntelliJ
 * Build and upload the model to the Local Maven Repo: `./gradlew -p ./model build publishToMavenLocal`
-* Enable Annotation processing (Shift+Shift and search for it)
 
-### Run the project
+
+### Run the Control Service using helm and k8s
+
+**Prerequisites**
+
+* Java 11 or greater - make sure JAVA_HOME is set
+* Docker latest - make sure "docker" is on the path
+* (for the helm charts) Helm 3+ - make sure that "helm" is on the path
+* (optional) AWS CLI configured with your account for the Taurus AWS account
+* Kerberos related functionality depends on shell and kadmin binary tool  (see Docker image)
+
+
+Build the project and create a docker image
+
+```bash
+  ./cicd/build.sh
 ```
-./gradlew :pipelines_control_service:docker  --info --stacktrace -Pversion=dev
-./gradlew :pipelines_control_service:bootRun
+
+Versatile Data Kit would work in a variety of Kubernetes environments. If you want to get started with a local deployment, we recommend kind.
+To deploy kind follow the `Deploy kubernetes cluster` and `Deploy local image registry` shown in the readme in ./data_pipelines_operator  without the `docker push`.
+
+To install the service, we will push its components to the Docker regsitry and deploy it through helm.
+
+From inside `` run:
+```
+aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin 012285273210.dkr.ecr.us-west-2.amazonaws.com
```

Comment:
Is ECR really still needed here? If it's left over let's remove it.

Reply:
Not sure, I haven't tried this part locally. Opened a ticket to check it and update the docs. 

https://github.com/vmware/versatile-data-kit/issues/1780